Gooliajulia/native signin

gooliajulia opened the pull request:
Refactors Sign In Screen to React Native

Notes:
- Will have to check the reroute to Edit Profile screen id params functionality
- Eventually substitute certain elements with our custom components when their ready (SingleActionButton, maybe Form)

yeezick committed:
laying out the landing screen and single action button

yeezick committed:
merging with newest udpates of native-development
Merge branch 'native-development' of https://github.com/yeezick/Bootcamper into yeezick/native-landing-screen

yeezick committed:
merging with native

yeezick committed:
- formatting client/SingleActionButton
- creating the native version of SingleActionButton with minimal styling and using switch cases to handle the variants
- clean up Landing and create payload-objects to pass into the button

yeezick committed:
apply custom functions to their respective case

yeezick committed:
removing a comment and typo

yeezick committed:
set up UserProfile skeleton which calls deployed api for a single user to render, then displays information about the user.

yeezick committed:
begin the transition for PortfolioCard
- this file should be broken up or renamed, 3 components live here and the name is mis leading
- missing form component conversion
- needs connection to passed props such as currProject

yeezick committed:
Merge pull request #54 from yeezick/yeezick/native-landing-screen

Yeezick/native landing screen

yeezick committed:
refactoring PortfolioCard to react-native, NOT considering functionality, only renderable components

yeezick committed:
- breaks apart PortfolioProject into 3 files
- all portfolio card components now react-native friendly

yeezick committed:
- breaks apart PortfolioProject into 3 files
- all portfolio card components now react-native friendly

yeezick committed:
includes UserPRofile as compatible component-wise

yeezick committed:
converting HEader, Form, and ShowPortfolioCard to react-native

yeezick committed:
handle path-related errors as a result of relocation, reorganize project/portfolio folders

yeezick committed:
- resolve path related errors in modal and form
- resolve 'cannot find document' error from htmlParser
- add button to edit profile

yeezick committed:
resolving errors regarding 'document' and removing nanoid

yeezick committed:
Form jsx and user/edit profiles has cohesive navigation now

yeezick committed:
single portfolio card is able to be edited

yeezick committed:
restoring handleTextChange to avoid rewriting state

yeezick committed:
remove console logs, pushing just to share drawer-navigation with nicholas

yeezick committed:
relocate sidemenu and reflect some changes in userprofile, pushing for nicholas

yeezick committed:
provide a solution that passes down props to drawer children

gooliajulia committed:
basic structure, functionality and styling refactored to native

gooliajulia committed:
remove commented out non native code

yeezick committed:
- replace nanoid with react-native-uuid
- slight restructure to SingleActionButton payload
- rename SinglePortfolioCard to Project AND formhandlers.js to just handlers
- provide slightly more semanticism to functions
- fully funcitoning CRUD for user info and portfolio projects

yeezick commented on native/screens/SignIn.jsx:
```diff
@@ -1,103 +1,166 @@
-//     password: "test",
-//   });
+export const SignIn = ({navigation}) => {
+  // redux
```
same comment as SignUp PR, let your code speak for you

yeezick commented on native/screens/SignIn.jsx:
```diff
@@ -1,103 +1,166 @@
-//     signOut();
-//   }, []);
+  // sign out user on sign in screen render
+  useEffect(() => {
```
not necessary, ideally a logged-in user should never see the SignIn screen again unless they log themselves out

yeezick commented on native/screens/SignIn.jsx:
```diff
@@ -1,103 +1,166 @@
+      });
+      // check this params functionality on edit land
+      navigation.navigate('EditProfile', {
+        id: user._id,
```
same comment as sign up; semanticism

yeezick commented on native/screens/SignIn.jsx:
```diff
@@ -1,103 +1,166 @@
-// };
+const styles = StyleSheet.create({
+  accountForms: {
+    flex: 1,
```
same comment as sign up; alphabetization

yeezick commented on native/screens/SignIn.jsx:
```diff
@@ -1,103 +1,166 @@
-//   };
+  const handleSignIn = async () => {
+    await dispatch(loginUser(logInInfo));
+    const user = await verify();
```
curious about this because we are dispatching the login info before we verify that their information is correct
if it works, it works, i'm curious about if there may be a better solution. ping me if you'd like to put our heads together on it

gooliajulia replied:
When I try to use the user._id from useSelector to reroute to EditProfile, the state is not updated after dispatching loginUser.

I've been using verify here to get the user id via the token since that proves to be updated appropriately during login and returns the most recent signed in user.

I've done some investigating into this and have had trouble getting a clear understanding of a solution, but maybe your understanding of redux would help clean some things up.

yeezick replied:
I just finished looking into this. Since React waits for the function execution to finish before updating the redux state, we won't be able to call the API, update the state, then check it in the same function. None of that information would be available until the next render.

Therefore, i deprecated that `loginUser` function in redux and used the actual `signIn`  function from `services/api` so that we can actually verify the response from the api before we update the redux store. It was a weird React-state lifecycle thing but good initial solution!

yeezick commented on native/screens/SignIn.jsx:
```diff
@@ -1,103 +1,166 @@
+    await dispatch(loginUser(logInInfo));
+    const user = await verify();
+    if (user.email === logInInfo.email) {
+      setLogInInfo({
```
since the user will be navigated to another screen, we can save a few lines by avoiding the need to reset the form info. if you disagree, let me know!

gooliajulia replied:
That makes sense, I added this in thinking more short term/ the current state of the app so that when returning to the sign in screen for testing purposes, the form is reset. 

But like you mentioned earlier, ideally a user will not even see/return to the Sign In screen unless they explicitly log out. I'll remove this now and we can revisit where this form reset logic will live according to logout and sign in screen redirect etc.

yeezick commented:
changed the base branch to be the branch that this is an extension of... 
the ... is because the hamster wheel in my head is turning to figure out if that is best practice or not.... maybe not.... lol

yeezick committed:
Merge pull request #58 from yeezick/yeezick/native-user-edit-profile

Yeezick/native user edit profile

yeezick committed:
pulling down changes from current native0development to resolve conflicts and use latest code
Merge branch 'native-development' of https://github.com/yeezick/Bootcamper into gooliajulia/native-signin

yeezick committed:
- update root useEffect to set up redux store on app render based on user verification
- remove comments from SignIn
- configure sign in to make the api call itself instead of thru redux-async-dispath method due to poor error handling and lack of tangible response from that method, it is now deprecated
- update signIn api function call to handle unsuccessful login attempts better
- replace onChangeText function with pre-built handler
- added error handling for empty payload in redux updateUser

yeezick committed:
alphabetize stylsheet
